SIL.Windows.Forms.Keyboarding/Windows/WinKeyboardAdaptor.cs

```diff
@@ -73,18 +73,21 @@ private static ushort HklToLangId(IntPtr hkl)
 		{
 			return (ushort)((uint)hkl & 0xffff);
 		}
+
 		public void UpdateAvailableKeyboards()
 		{
 			var curKeyboards = KeyboardController.Instance.Keyboards.OfType<WinKeyboardDescription>().ToDictionary(kd => kd.Id);
 
 			foreach (InputLanguage inputLanguage in InputLanguage.InstalledInputLanguages)
 			{
-				var keyboardId = $"{inputLanguage.Culture.Name}_{inputLanguage.LayoutName}";
-				var keyboardLayoutName = GetBestAvailableKeyboardName(inputLanguage);
+				string keyboardId;
+				LayoutName keyboardLayoutName;
 				CultureInfo culture;
 				string cultureName;
 				try
 				{
+					keyboardId = $"{inputLanguage.Culture.Name}_{inputLanguage.LayoutName}";
```

Comment:
I'd keep the same order as in the catch block:

    culture =
    cultureName = 
    keyboardId =
    keyboardLayoutName =

and use `culture.Name` for setting `keyboardId`.